components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/base/base_container.html

```diff
@@ -195,6 +195,61 @@
         {% if page_size %}
             var PAGE_SIZE = {{ page_size }};
         {% endif %}
+
+
+        // ** "Open With" config used by E.g. ome.tree.js **
+        WEBCLIENT.OPEN_WITH = [];
+        {% if ome.open_with %}
+            // This is a json dump, so we can add it directly without escaping
+            WEBCLIENT.OPEN_WITH = {{ ome.open_with|safe }};
+        {% endif %}
+        // Helper can be used by 'open with' plugins to add isEnabled()
+        // handlers to the OPEN_WITH object.
+        OME.setOpenWithEnabledHandler = function(label, fn) {
+            // look for label in OPEN_WITH
+            WEBCLIENT.OPEN_WITH.forEach(function(ow){
+                if (ow.label === label) {
+                    ow.isEnabled = function() {
+                        // wrap fn with try/catch, return true by default
+                        var args = Array.from(arguments);
+                        try {
+                            enabled = fn(args);
+                        } catch (e) {
+                            console.log("Open with " + label + ": " + e);
```

Comment:
@will-moore  makes as good point regarding the "communication" of the error.

I think the console message here is the only way really how a user can (when pointed towards the dev tools) share the error, given that it has to be caught and that the browser does not give you a 'backend type of log', the only way one could achieve that would be a logging service with ajax requests sent to it.

I realize, however, that console or any of its methods (despite all major browsers supporting it in their more recent versions - according to MDN) can be potentially undefined but a check for that can address that issue. I would use it very sparingly, this one case seems a potential candidate.